docs/internal/input-validations-framework-readme.md

```diff
@@ -0,0 +1,75 @@
+# How to use the framework
+
+The constraints framework implementation is located in the utils/constraints-utils.ts module
+
+import * from 'utils/constraints-utils';
+
+## What can you use with the framework
+
+You will find in the constraints framework the generic calls for the following classes if you properly import them from the constraints framework.
+
+## generic class of validations constraints-utils.ts
+
+This class holds the supported constraints and function(s) to validate constraints defined below in the rest of this document.
+
+## StringConstraint
+
+Constructor: new StringConstraint(minValue, maxValue);
+
+API reference ['here'](/docs/api/classes/utils.StringConstraint.html)
+
+If given string length falls outside of these inclusive bounds throws detailed Zod error
+
+## UrlStringConstraint
+
+Constructor: new UrlStringConstraint(minValue, maxValue); 
+
+API reference ['here'](/docs/api/classes/utils.UrlStringConstraint.html)
+
+If given string length falls outside of these inclusive bounds, or does not follow a proper URL format it throws detailed Zod error
+
+## NumberConstraint
+
+Constructor: new NumberConstraint(minValue, maxValue);
+
+API reference ['here'](/docs/api/classes/utils.NumberConstraint.html)
+
+If given number falls outside of these inclusive bounds throws detailed Zod error.
+
+## ArrayConstraint
+
+new utils.ArrayConstraint(minValue, maxValue);
+
+API reference ['here'](/docs/api/classes/utils.ArrayConstraint.html)
+
+If given array length falls outside of these inclusive bounds throws detailed Zod error.
+
+## validateConstraints Function
+
+validateConstraints<T>(constraints: ConstraintsType<T>, context: string, ...object: any) is the entry point to use the framework. This function can validate either a single object or an array of objects against the provided constraints.
+
+## How to use the constraints-utils.ts
+
+You need two things when utilizing constraints-utils.ts and the following examples are from ['here'](/lib/stacks/eks-blueprint-stack.ts)
```

Comment:
This sentence does not read right, it is combining two unrelated things. 

When you say "you need two things" either you tell what they are or below you have a numbered list. E.g. first you need to define constraints object  and second ...